### PR TITLE
TextConverter: Fix hex/bin/base64 conversion bugs.

### PR DESCRIPTION
**Pull request by pjhampton**
## Description of new Instant Answer, or changes

Fixes the conversion errors with hex/bin/base64. 

screen shot 2017-08-10 at 11 02 40 pm

## Related Issues and Discussions

Fixes: https://github.com/duckduckgo/zeroclickinfo-goodies/issues/4411

## People to notify

@moollaza 

## Testing & Review
To be completed by Community Leader (or DDG Staff) when reviewing Pull Request

**Pull Request**
- [x] Title follows correct format (Specifies Instant Answer + Purpose)
- [x] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Code**
- [ ] Adheres to the [DuckDuckGo Style Guide](https://docs.duckduckhack.com/resources/code-style-guide.html)
- [ ] Behaviour is appropriately tested. If improvement, tests are adequately extended.
- [ ] There is no unnecessary files in place (such as editor config files)
- [ ] There is no API keys / secrets present
- [ ] Tests are passing (run `$ duckpan test <goodie_id>`)
    - Tester should report any failures

**Ready to merge?**

- [ ] Has this IA been deployed to and tested on [beta.duckduckgo.com](https://beta.duckduckgo.com/)?
- [ ] For larger commits, has this been approved be more than one community member?
- [ ] The number of reviews is appropriate for this type of PR
- [ ] The commit message is clear, coherent and fitting

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/text_converter

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/goodie/text_converter/text_converter.js [hunk 1]
```diff
@@ -167,36 +252,63 @@ DDH.text_converter = DDH.text_converter || {};
...
+                    hex_result += result.charAt(j);
                 } 
             }
-            return pretty_result;
+            return hex_result;
         },
 
-        hexToText: function( hex ) {
+        hexToText: function(hex) {
             return hex.split(/\s/).map(function (val) {
```
Comment (moollaza) [inline share/goodie/text_converter/text_converter.js]: If we detect that the input string has no spaces, and is more than 2 chars, we can split it into groups of 2 characters and convert it to text.

Should allow a non-padded input like `0x5468697320697320612074657374` to be converted 👍

**Comment by moollaza** [comment]
@pjhampton Thanks! This LGTM :+1:

**Comment by bsstoner** [comment]
This was released to production 🚀

**Pull request merged**